opensearch-operator/pkg/reconcilers/cluster.go

```diff
@@ -145,6 +145,18 @@ func (r *ClusterReconciler) reconcileNodeStatefulSet(nodePool opsterv1.NodePool,
 	if err != nil {
 		return result, err
 	}
+
+	// Fix selector.matchLabels (issue #311), need to recreate the STS for it as spec.selector is immutable
```

Comment:
Thats kind of an HotFix to running clusters ?

Reply:
Exactly. As the selector is immutable recreating the STS is the only way to apply the change. And leaving it the old way would leave the bug open for existing clusters.

Reply:
that's a good implementation, did we check it on a running cluster? I dont want to marge it and then create problem on running clusters

Reply:
Besides that, all looks good and ready to merge

Reply:
I tested several scenarios on a local k3s cluster and they worked without problems.